src/client.rs

```diff
@@ -1,269 +1,24 @@
-    fn to_string(&self) -> String {
-        let total_seconds = self.0.as_secs();
-        let days = total_seconds / 86400;
-        let hours = (total_seconds % 86400) / 3600;
-        let minutes = (total_seconds % 3600) / 60;
-        let seconds = total_seconds % 60;
-
-        format!("{days} days, {hours}:{minutes:02}:{seconds:02}")
-    }
-}
-#[derive(Serialize)]
-struct Stats {
-    artists: u32,
-    albums: u32,
-    songs: u32,
-    uptime: String,
-    playtime: String,
-    db_playtime: String,
-    db_update: String,
-}
-
-impl Stats {
-    pub fn new(stats: mpd::stats::Stats) -> Self {
-        let seconds: i64 =
-            stats.db_update.as_secs().try_into().unwrap_or(i64::MAX);
-        let db_update = match Utc.timestamp_opt(seconds, 0) {
-            chrono::LocalResult::Single(date_time) => {
-                date_time.format("%a %b %d %H:%M:%S %Y").to_string()
-            }
-            _ => String::new(),
-        };
-
-        Self {
-            artists: stats.artists,
-            albums: stats.albums,
-            songs: stats.songs,
-            uptime: HumanReadableDuration::from(stats.uptime).to_string(),
-            playtime: HumanReadableDuration::from(stats.playtime).to_string(),
-            db_playtime: HumanReadableDuration::from(stats.db_playtime)
-                .to_string(),
-            db_update,
-        }
-    }
-}
-
-impl fmt::Display for Stats {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(
-            f,
-            "artists={}\nalbums={}\nsongs={}\nuptime={}\nplaytime={}\ndb_playtime={}\ndb_update={}",
-            self.artists,
-            self.albums,
-            self.songs,
-            self.uptime,
-            self.playtime,
-            self.db_playtime,
-            self.db_update,
-            )
-    }
-}
-
-#[derive(Serialize)]
-pub struct Playlists {
-    #[serde(serialize_with = "serialize_playlists")]
-    playlists: Vec<Playlist>,
-}
-
-#[derive(Default, Serialize)]
-pub struct Playlist {
-    pub name: String,
-    songs: Vec<Song>,
-}
-
-impl From<String> for Playlist {
-    fn from(name: String) -> Self {
-        Playlist {
-            name,
-            ..Default::default()
-        }
-    }
-}
-
-impl fmt::Display for Playlists {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        for (index, playlist) in self.playlists.iter().enumerate() {
-            write!(f, "{}={}", index, playlist.name)?;
-        }
-
-        Ok(())
-    }
-}
-
-#[derive(Serialize)]
-pub struct Outputs {
-    outputs: Vec<Output>,
-}
-
-#[derive(Serialize)]
-pub struct Output(String);
-
-impl From<String> for Output {
-    fn from(name: String) -> Self {
-        Output(name)
-    }
-}
-
-impl fmt::Display for Outputs {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        for (index, output) in self.outputs.iter().enumerate() {
-            write!(f, "{}={}", index, output.0)?;
-        }
-
-        Ok(())
-    }
-}
```

Comment:
There was a fair number of new structs introduced here so I took the opportunity to extract some modules.